auth_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAuthManager(t *testing.T) {
+	ms := NewMemStore()
+
+	cmdr := newTestOSCalls("mac=foobar", 0)
+	am := NewAuthManager(ms, "devkey", IdentityDataRunner{
+		cmdr: &cmdr,
+	})
+	assert.IsType(t, &MenderAuthManager{}, am)
+
+	mam := am.(*MenderAuthManager)
+
+	assert.False(t, am.HasKey())
+	assert.NoError(t, am.GenerateKey())
+	assert.True(t, am.HasKey())
+
+	assert.False(t, am.IsValid())
+
+	code, err := am.AuthCode()
+	assert.Equal(t, noAuthCode, code)
+	assert.NoError(t, err)
+
+	ms.WriteAll(authTokenName, []byte("footoken"))
+	code, err = am.AuthCode()
+	assert.Equal(t, AuthCode("footoken"), code)
+	assert.NoError(t, err)
+
+	_, err = am.MakeAuthRequest()
+	assert.Error(t, err, "should fail, no tenant token is present")
+
+	// setup tenant token
+	ms.WriteAll(authTenantTokenName, []byte("tenant"))
+	// setup sequence number
+	ms.WriteAll(authSeqName, []byte("12"))
+
+	req, err := am.MakeAuthRequest()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, req.Data)
+	assert.Equal(t, AuthCode("tenant"), req.Code)
+	assert.NotEmpty(t, req.Signature)
+
+	var ard AuthReqData
+	err = json.Unmarshal(req.Data, &ard)
+	assert.NoError(t, err)
+
+	assert.Equal(t, AuthReqData{
+		IdData:      "{\"mac\":\"foobar\"}",
+		TenantToken: "tenant",
+		Pubkey:      mam.keystore.PublicPEM(),
+		SeqNumber:   13,
+	}, ard)
+
+	sign, err := mam.keystore.Sign(req.Data)
+	assert.Equal(t, sign, req.Signature)
+
+	err = am.RecvAuthResponse([]byte{})
+	// should fail with empty response
+	assert.Error(t, err)
+
+	// make storage RO
+	ms.ReadOnly(true)
+	err = am.RecvAuthResponse([]byte("fooresp"))
+	assert.Error(t, err)
+
+	ms.ReadOnly(false)
+	err = am.RecvAuthResponse([]byte("fooresp"))
+	tokdata, err := ms.ReadAll(authTokenName)
+	assert.NoError(t, err)
+	assert.Equal(t, []byte("fooresp"), tokdata)
+}
```

Comment:
I would split it into multiple tests. This is quite a big test and could be hard to figure out what is broken once test will fail.